File: typescript/src/main/vsdk/toolkit/io/context/TransformStackContext.ts
import { TransformContext } from "./TransformContext";
import { TransformSequenceContext } from "./TransformSequenceContext";

export class TransformStackContext {
  public xid: number;
  public xac: number;
  public rev: number;
  public ownedArgumentCount: number;
  public xf: TransformContext;
  public transformationArray: TransformSequenceContext | null;
  public ownedArgumentCopies: Array<string | null> | null;
  public prev: TransformStackContext | null;

  public constructor() {
    this.xid = 0;
    this.xac = 0;
    this.rev = 0;
    this.ownedArgumentCount = 0;
    this.xf = new TransformContext();
    this.transformationArray = null;
    this.ownedArgumentCopies = null;
    this.prev = null;
  }
}

File: java/base/src/main/java/vsdk/toolkit/io/context/TransformStackContext.java
package vsdk.toolkit.io.context;

public class TransformStackContext {
    public long xid; // Unique transform id
    public short xac; // Context argument count
    public short rev; // Boolean true if vertices reversed
    public short ownedArgumentCount; // Number of owned argument copies
    public TransformContext xf; // Cumulative transformation
    public TransformSequenceContext transformationArray;
    public String[] ownedArgumentCopies; // Copies for non-iterative transform arguments
    public TransformStackContext prev; // Previous transformation context

    public TransformStackContext() {
        xid = 0;
        xac = 0;
        rev = 0;
        ownedArgumentCount = 0;
        xf = new TransformContext();
        transformationArray = null;
        ownedArgumentCopies = null;
        prev = null;
    }
}
